projects/wave-core-new/src/lib/layers/layer.model.ts

```diff
@@ -4,13 +4,18 @@ import {Unit} from '../operators/unit.model';
 
 export type LayerType = 'raster' | 'vector';
 
-export abstract class Layer implements ToDict<LayerDict> {
-    protected _name: string;
+export abstract class Layer implements HasLayerId, HasLayerType, ToDict<LayerDict> {
+    protected static nextLayerId = 0;
+
+    abstract readonly layerType: LayerType;
+
+    readonly id: number;
+
+    readonly name: string;
     readonly workflowId: UUID;
 
-    // TODO: move to layer service
-    protected _isVisible = true;
-    protected _isLegendVisible = false;
+    readonly isVisible: boolean;
+    readonly isLegendVisible: boolean;
```

Comment:
I guess the legend state could also live in the layerlist. However, this is a topic for later discussion.

Reply:
Yeah, this is a separate task.